File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cs
public class Solution {
    public bool IncreasingTriplet(int[] nums) {
        int x = int.MaxValue;
        int y = int.MaxValue;
        foreach (int it in nums) {
            if (it <= x)
                x = it;
            else if (it <= y)
                y = it;
            else
                return true;
        }
        return false;
    }
}

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int x=INT_MAX,y=INT_MAX;
    for (auto& it:nums){
        if (it<=x)
            x=it;
        else if (it<=y)
            y=it;
        else
            return true;
    }
        return false;
    }
};